cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -264,6 +264,13 @@ macro(swift_common_unified_build_config product)
   endif()
 endmacro()
 
+# Common cmake project config for compilers.
+macro(swift_common_cxx_compiler_setup)
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
+    set(SWIFT_COMPILER_IS_MSVC_LIKE TRUE)
+  endif()
+endmacro()
```

Comment:
A comment would be nice, too - something like "are we being built with an MSVC-like compiler?". It just makes it easier to follow, rather than trying to decode and interpret the variable name.